config/7.1.0/securebanking/ig/config/dev/config/config.json

```diff
@@ -69,18 +69,62 @@
       "name" : "ForgeRockClientHandler",
       "type" : "Chain",
       "config" : {
-        "filters" : [ "TransactionIdOutboundFilter" ],
+        "filters" : [ 
+          {
+            "comment": "Add x-fapi-interaction-id header if one was not present in the request",
+            "name": "FapiInteractionIdFilter",
+            "type": "FapiInteractionIdFilter"
+          },
+          {
+            "comment": "Copy the x-fapi-interaction-id header to TransactionIdContext",
+            "name": "FapiTransactionIdFilter-1",
+            "type": "ScriptableFilter",
+            "config": {
+              "type": "application/x-groovy",
+              "file": "FapiTransactionIdFilter.groovy"
+            }
+          },
```

Comment:
I think each route's filter chain should start with these 2 filters. These filters should then be removed from the `ForgeRockClientHandler` and `SBATReverseProxyHandler`.

It makes sense to add the `TransactionIdOutboundFilter` to the `SBATReverseProxyHandler` (which you have done). I think the `ForgeRockClientHandler` should be reverted to it's previous state.

Both of the filters need to be at the start of each route's chain in order to get the transactionId into the context as earlier as possible, this will then be passed down to other FR products either by the FRClientHandler or the RPHandler. Adding them to the handler's themselves is too late, other filters will have already run and potentially need ed to log the value.

It is unfortunate to have to duplicate the config across all routes, but it seems the best way to go to me.

Reply:
Yeah, I think you are right Dan. Seems a waste to have to duplicate. I could create a chain with both in and then only one definition in each route.

Reply:
Ah! https://backstage.forgerock.com/docs/ig/7.2/reference/ChainOfFilters.html